### ISSUE TITLE
[QUESTION] How are Queries sorted in v2?

### ISSUE THREAD

**Issue by IMBurbank** [dynamoose/dynamoose#855]
### Summary:
I'm testing our app with the major version upgrade to v2.1.0. We were previously sorting queries before execution with `query.ascending()` and `query.descending()`; 

I can no longer find these methods - on the query object, or in the documentation.

How do we sort queries in v2?

#### Schema
```javascript
const dynamoose = require('dynamoose');
const Schema = new dynamoose.Schema({
    id: { type: String, hashKey: true, required: true },
    createdAt: { type: Number, rangeKey: true, default: Date.now },
    createdBy: { type: String, required: true },
    schemaName: { type: String, default: "fooSchema" },
    transactionType: { type: String, required: true },
});
```

#### Model
```javascript
const schemaTableName = "tableFoo";

const schemaConfig = {
    throughput: "ON_DEMAND",
    serverSideEncryption: true,
    create: true,
    update: false,
    waitForActive: true,
    waitForActiveTimeOut: 10000,
    defaultReturnValues: "ALL_NEW"
};

module.exports = dynamoose.model(schemaTableName, Schema, schemaConfig);
```

#### General
```javascript
const Model = require('./path/to/model');
const f = { id: { eq: "foo" }};
const query = Model.query(f);
const sorted = query.descending();
// Error: query.descending is not a function
```

### Environment:

Operating System: osX
Operating System Version: Mojave 10.14.6
Node.js version (`node -v`): 12.16.2
NPM version: (`npm -v`): 6.14.4
Dynamoose version: 2.1.0

### Other information (if applicable):
NA

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this question has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this question hasn't been raised or answered before
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by IMBurbank** [comment]
Related, but not the same as #848

**Comment by fishcharlie** [comment]
@IMBurbank This actually is a duplicate of #848.

https://github.com/dynamoose/dynamoose/blob/9e5fb1605e99da2d5550cfa896d224c09aaa3cc8/lib/Query.js#L207-L209

In v1 as you can see above, when you do `descending` it sets `ScanIndexForward`.

**Comment by MickL** [comment]
Actually the exact same as #848 ☺️

**Comment by jdrydn** [comment]
Ah, just spent ages trying to debug:

```
Model.query(...).sort is not a function
```

But turns out I wasn't on the latest (`2.2.0` in this case) - may be worth reflecting this in the documentation? "Released in `2.2.0`" or something similar?

**Comment by fishcharlie** [comment]
@jdrydn Hmm. I'm not sure how to handle this. Because on one hand, it is valuable to know when something is released. However, at some point it becomes no longer relevant. If everyone is using newer versions, that just adds bloat to the documentation, and overtime becomes no longer relevant to people viewing the documentation. For example, years from now, that comment becomes useless. And having to maintain it would be a huge pain. Unless we can figure out a way to automate it somehow.

It might be worth considering using [versioning](https://v2.docusaurus.io/docs/versioning). However this only seems useful between major versions (ie. v2 to v3), where there are breaking changes. And they encourage you to think about if versioning is correct in your use case. Since it does add a **lot** of bloat.

I'm curious if you happened to look at either #848 or #860 before coming here. At some point I do want to build a system where when a release goes out, it comments on all relevant pull requests (and **MAYBE** issues) with which release it was part of. Because issues do get closed and pull requests get merged that aren't part of a release. So having a system automatically comment on things with that relevant information would be super cool. But of course in this case it would have only helped if you looked at either #848 or #860 early in your debugging process. Duplicates like this make it more difficult.

---

I will say also. If you are spending ages trying to debug something. Contact me! I'd love to help. Our bug report template includes a field for you to fill out what version of Dynamoose you are running. And I could have told you in an instant that you need to upgrade, since I'm aware that it was **just** released. And even if you contacted me on the Slack channel or something my first question would have been what version of Dynamoose were you running? Since it was released so recently. So contact me before banging your head against a wall for a long period of time. I can try to help as much as I can!!
